sound/soc/sof/intel/hda-loader.c

```diff
@@ -558,8 +585,8 @@ int hda_dsp_ipc4_load_library(struct snd_sof_dev *sdev,
 
 	/* prepare DMA for code loader stream */
 	hext_stream = hda_cl_prepare(sdev->dev, HDA_CL_STREAM_FORMAT,
-				     stripped_firmware.size,
-				     &dmab, SNDRV_PCM_STREAM_PLAYBACK, false);
+				     stripped_firmware.size, &dmab, false,
```

Comment:
don't we also need to keep this buffer permanently? 

Not sure I understand why the firmware buffer would be kept but the library would be treated differently.

Reply:
The library (and iccmax capture?) buffer is not yet handled, the issues so far only been hit with older platforms running ipc3, hence no libraries nor iccmax.
iccmax should be relatively easy while the library is not as we don't know how many of them we have..

Reply:
@ujfalusi are you saying that we need N buffers for N libraries? That would obviously be problematic.

One would think there's a way to reuse a single worst-case buffer for all this, at the end of the day the sum of base firmware + N libraries cannot exceed the size of the internal memory, can it?

Reply:
@plbossart, hrm, right, I guess we could try to get 'big enough' buffer and just do memcpy to it.
Allocate for basefw first and if any of the library needs more space then re-allocate a bigger buffer and keep using that.
Probably I would have arrived to this in few days ;)